app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,19 @@
+class SessionsController < ApplicationController
+  def new; end
+
+  def create
+    user = User.find_by(email: params[:session][:email].downcase)
+    if user && user.authenticate(params[:session][:password])
```

Comment:
dùng user&. đi chú, tìm hiểu công dụng của nó néh

Reply:
Toán tử &. ko hoàn toàn tương đồng về kết quả với việc sử dụng && trong 1 số ít trường hợp. Ví dụ trong trường hợp này, mục đích của ta là trả về false hoặc true thì việc sử dụng 2 toán tử trên là tương đương. 
Ngoài ra sử dụng &. khiến code ngắn gọn và dễ nhìn

Reply:
trong trường hợp nào thì nó không tương đồng thế chú ?

Reply:
GIả sử account có owner, owner có address.
Sử dụng if account && account.owner&&account.owner.address và account&.owner&.address sẽ khác nhau trong trường hợp:
Nếu owner = false thì && trả về nil còn &. trả về  undefined method 'address' for False:FalseClass.

Reply:
@Tungdaigia  sai sai nhỉ ?
owner = false ?
nếu owner là boolean true false thì sẽ khôgn có address đằng sau 
còn owner là nil thì không có vụ false:false class gì gì đó kia 
và acctount&...&... hoàn toàn chạy đc ?

Reply:
vậy là 2 cách viết là giống nhau ạ

Reply:
ready